packages/foam-vscode/src/features/folder-rename-handler.ts

```diff
@@ -0,0 +1,1112 @@
+import * as vscode from 'vscode';
+import { Foam } from '../core/model/foam';
+import { FoamWorkspace } from '../core/model/workspace';
+import { MarkdownLink } from '../core/services/markdown-link';
+import { Logger } from '../core/utils/log';
+import { isAbsolute, getDirectory, relativeTo } from '../core/utils/path';
+import { getFoamVsCodeConfig } from '../services/config';
+import { fromVsCodeUri, toVsCodeRange, toVsCodeUri } from '../utils/vsc-utils';
+import {
+  FolderRenameDialog,
+  DryRunCalculationResult,
+  FolderRenameDialogResult,
+} from './folder-rename-dialog';
+import * as path from 'path';
+
+interface FolderRenameResult {
+  filesProcessed: number;
+  linksUpdated: number;
+  errors: string[];
+  warnings: string[];
+}
+
+interface FolderRenameOptions {
+  confirmAction?: boolean;
+}
+
+interface ProcessFilesInternalResult {
+  linksUpdatedCount: number;
+  filesProcessedCount: number;
+  calculationErrors: string[];
+}
+
+/**
+ * Handles folder renames with comprehensive link updates
+ */
+export class FolderRenameHandler {
+  private foam: Foam;
+  constructor(foam: Foam) {
+    this.foam = foam;
+  }
+
+  /**
+   * Main entry point for handling folder rename events
+   */
+  async handleFolderRename(
```

Comment:
this function is a bit long and does a few things, do you think it would be possible to break it down into sub-functions for the various steps?